src/abstractdataframe/io.jl

```diff
@@ -1,17 +1,64 @@
+"""
+    show(io::IO, mime::MIME, df::AbstractDataFrame)
+
+Render a data frame to an I/O stream in MIME type `mime`.
+
+# Arguments
+- `io::IO`: The I/O stream to which `df` will be printed.
+- `mime::MIME`: supported MIME types are: `"text/plain"`, `"text/html"`, `"text/latex"`,
+  `"text/csv"`, `"text/tab-separated-values"`
+- `df::AbstractDataFrame`: The data frame to print.
+
+Additionally selected MIME types support passing the following keyword arguments:
+- MIME type `"text/plain"` accepts all listed keyword arguments and therir behavior
+  is identical as for `show(::IO, ::AbstractDataFrame)`
+- MIME type `"text/html"` accepts `summary` keyword argument which
+  allows to choose whether to print a brief string summary of the data frame.
+
+# Examples
+```jldoctest
+julia> show(stdout, MIME("text/latex"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+\\begin{tabular}{r|cc}
+        & A & B\\\\
+        \\hline
+        & Int64 & String\\\\
+        \\hline
+        1 & 1 & x \\\\
+        2 & 2 & y \\\\
+        3 & 3 & z \\\\
+\\end{tabular}
+14
+
+julia> show(stdout, MIME("text/csv"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+"A","B"
+1,"x"
+2,"y"
+3,"z"
+```
+"""
```

Comment:
Actually since https://github.com/JuliaLang/IJulia.jl/issues/835 doesn't work it looks like you can attach the docstring to any signature, even if it doesn't correspond to a real method:
```suggestion
"""
Base.show(io::IO, mime::MIME, df::AbstractDataFrame)
```

Reply:
Can you? I would assume that such a statement would be treated by Julia compiler as a function call and say that `io`is an undefined variable?

Reply:
We could add this signature and define additional `showable` methods (this should not be overly problematic) if you prefer.

Reply:
I don't know how, but it works in my tests.

Reply:
OK - I will try committing your suggestion and see if it works.